src/cartographer/config/overlays/excluded-components.yaml

```diff
@@ -0,0 +1,30 @@
+# Copyright 2022 VMware
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:data", "data")
+
+#! Exclude resources for components that a user has chosen to not install as defined by
+#! the excluded_components property. Each entry in excluded_components defines a new
+#! overlay that removes resources with a label matching that value. Specifying a
+#! component that doesn't match any resources will error.
+
+#@ for exclude in data.values.excluded_components:
+
+#! TODO(scothis) drop "component" label once we've migrated to "app.kubernetes.io/component"
+#@overlay/remove
+#@overlay/match by=overlay.or_op(overlay.subset({"metadata":{"labels":{"component": exclude}}}), overlay.subset({"metadata":{"labels":{"app.kubernetes.io/component": exclude}}})), expects="1+"
```

Comment:
uhh neat!

Reply:
Cartographer conventions currently uses a label `component: conventions.carto.run` to identify resources that are part of that component, it would be great to normalize this between the core and conventions.

I'd propose both components use `app.kubernetes.io/component`as the label name. Being able to exclude the core would also require that the cartographer-system namespace resource be defined independently.

Reply:
👍 👍  

I imagine the next step would be to also add `app.kubernetes.io/component` to cartographer's objects, right? if so, happy to do

Reply:
cartographer: https://github.com/vmware-tanzu/cartographer/pull/794